### PR TITLE
Fix Up Typing in GroupBy

### PR DESCRIPTION
**Pull request by WillAyd**
More typing cleanups

Original failures:

```sh
pandas/core/groupby/ops.py:13: error: Module 'pandas._libs' has no attribute 'groupby'
pandas/core/groupby/ops.py:13: error: Module 'pandas._libs' has no attribute 'reduction'
pandas/core/groupby/groupby.py:22: error: Module 'pandas._libs' has no attribute 'groupby'
pandas/core/groupby/groupby.py:329: error: Need type annotation for '_apply_whitelist'
pandas/core/groupby/generic.py:220: error: Incompatible types in assignment (expression has type "Callable[[Arg(Any, 'arg'), VarArg(Any), KwArg(Any)], Any]", base class "SelectionMixin" defined the type as "Callable[[Arg(Any, 'func'), VarArg(Any), KwArg(Any)], Any]")
```

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -179,25 +179,25 @@ def _post_process_cython_aggregate(self, obj):
             obj = obj.swapaxes(0, 1)
         return obj
 
-    def aggregate(self, arg, *args, **kwargs):
+    def aggregate(self, func, *args, **kwargs):
```
Comment (WillAyd) [inline pandas/core/groupby/generic.py]: The reason for this change is that MyPy was complaining about the signature here differening from what's defined in `SelectionMixin`. This aligns the signatures and is arguably more readable.

Counter argument is that this is an API change, but this actually would match the docs

image

Comment (jreback) [reply pandas/core/groupby/generic.py]: this is ok, pls add a note in the whatsnew about this though

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -218,6 +218,7 @@ Other API Changes
 - :meth:`Timestamp.strptime` will now rise a ``NotImplementedError`` (:issue:`25016`)
 - Comparing :class:`Timestamp` with unsupported objects now returns :py:obj:`NotImplemented` instead of raising ``TypeError``. This implies that unsupported rich comparisons are delegated to the other object, and are now consistent with Python 3 behavior for ``datetime`` objects (:issue:`24011`)
 - Bug in :meth:`DatetimeIndex.snap` which didn't preserving the ``name`` of the input :class:`Index` (:issue:`25575`)
+- The ``arg`` argument in :meth:`pandas.core.groupby.DataFrameGroupBy.agg` has been renamed to ``func`` to match the signature of its superclass (:issue:`26089`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: I would remove the work superclass; this is unknown to the user

**Review by jreback — CHANGES_REQUESTED** [review]
merge master as well

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @WillAyd

**Pull request merged**